flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1963,7 +2025,9 @@ public void startCheckpointScheduler() {
             stopCheckpointScheduler();
 
             periodicScheduling = true;
-            currentPeriodicTrigger = scheduleTriggerWithDelay(getRandomInitDelay());
+            long initDelay = getRandomInitDelay();
+            nextCheckpointTriggeringRelativeTime = clock.relativeTimeMillis() + initDelay;
```

Comment:
Would it be simpler to update `nextCheckpointTriggeringRelativeTime` inside `scheduleTriggerWithDelay()`?